src/features/grading/GradingTypes.ts

```diff
@@ -23,8 +23,10 @@ export type GradingOverview = {
   currentXp: number;
   maxXp: number;
   studentId: number;
-  studentName: string;
-  studentUsername: string;
+  studentName: string | undefined;
```

Comment:
What if we use an intersection with a discriminated union to capture the constraint? Would that work?